sotopia/cli/benchmark/benchmark.py

```diff
@@ -78,21 +81,66 @@ def initilize_benchmark_combo(data: list[dict[str, str]]) -> list[EnvAgentComboS
     return list_of_env_agent_combo_storage
 
 
-def get_avg_reward(episodes: list[EpisodeLog], model_name: str) -> dict[str, float]:
-    rewards_list = []
+def get_avg_reward(
+    episodes: list[EpisodeLog], model_name: str
+) -> dict[str, tuple[float, float]]:
+    # rewards_list = []
+    rewards_dict = defaultdict(
+        list
+    )  # {pk: [rewards]}, {pk}_{i} denotes the i-th agent is the test agent
     avg_reward_dict = {}
+    avg_variance_dict = {}
+    var_reward_list = []
     for episode in episodes:
         assert episode.models is not None, "episode.models should not be None"
         if episode.models[1] == model_name:
             reward = get_rewards_from_episode(episode)[0][1]
+            rewards_dict[f"{episode.environment}_0"].append(reward)
         else:
             reward = get_rewards_from_episode(episode)[1][1]
-        rewards_list.append(reward)
+            rewards_dict[f"{episode.environment}_1"].append(reward)
+        # rewards_list.append(reward)
+    dimensions = list(rewards_dict.values())[0][0].keys()
+
+    def calc_variance(local_rewards_list: List[Dict[str, float]]) -> Dict[str, float]:
+        # get the variance within a list
+        local_var_reward_dict = {}
+        local_dimensions = local_rewards_list[0].keys()
+        assert set(local_dimensions) == set(dimensions), "dimensions should be the same"
+        for dimension in local_dimensions:
+            rewards = [reward[dimension] for reward in local_rewards_list]
+            avg_reward = sum(rewards) / len(rewards)
+            variance = sum([(reward - avg_reward) ** 2 for reward in rewards]) / len(
+                rewards
+            )
+            local_var_reward_dict[dimension] = variance
+
+        return local_var_reward_dict
+
+    rewards_list = list(chain(*rewards_dict.values()))
+
+    var_reward_list = [calc_variance(rewards) for rewards in rewards_dict.values()]
     for dimension in rewards_list[0].keys():
         rewards = [reward[dimension] for reward in rewards_list]
         avg_reward = sum(rewards) / len(rewards)
         avg_reward_dict[dimension] = avg_reward
-    return avg_reward_dict
+
+        variances = [variance[dimension] for variance in var_reward_list]
+        # print(f"Dimension {dimension}, variances: {variances}, {len(variances)}")
+        avg_variance = sum(variances) / len(variances)
+        avg_variance_dict[dimension] = avg_variance
+
+    return_rewards_dict = {
+        key: (avg_reward_dict[key], math.sqrt(avg_variance_dict[key]))
+        for key in avg_reward_dict.keys()
+    }
+    return_rewards_dict = {
+        **return_rewards_dict,
+        "setting_num": (float(len(var_reward_list)), 0.0),
+        "episode_count": (float(len(rewards_list)), 0.0),
+    }
+
+    return return_rewards_dict
```

Comment:
I notice we are doing thing wrong to calculate the CI bound: @bugsz please follow here 
![image](https://github.com/user-attachments/assets/dc570c55-90f9-41f2-86a5-d8b6ccebb69e)

Reply:
Update: Now I am 
1. getting the variances within each scenario  (setting) then average them for an variance estimation. 
3. calculate the 95% CI bound